ui/packages/tidb-dashboard-lib/src/apps/UserProfile/components/Form.SSO.tsx

```diff
@@ -252,40 +252,47 @@ export function SSOForm() {
                   label={t('user_profile.sso.form.client_id')}
                   rules={[{ required: true }]}
                 >
-                  <Input
-                    disabled={!isWriteable}
-                    style={DEFAULT_FORM_ITEM_STYLE}
-                  />
-                </Form.Item>
-                <Form.Item
-                  name="client_secret"
-                  label={t('user_profile.sso.form.client_secret')}
-                  rules={[{ required: false }]}
-                >
-                  <Input
-                    disabled={!isWriteable}
-                    style={DEFAULT_FORM_ITEM_STYLE}
-                  />
-                </Form.Item>
-                <Form.Item
-                  name="scopes"
-                  label={t('user_profile.sso.form.scopes')}
-                  rules={[{ required: false }]}
-                >
-                  <Input
-                    disabled={!isWriteable}
-                    style={DEFAULT_FORM_ITEM_STYLE}
-                    placeholder="openid profile email"
-                  />
+                  <Input disabled={!isWriteable} style={{ width: 320 }} />
                 </Form.Item>
+                {
+                  // to compatible with old version
+                  config?.client_secret !== undefined && (
+                    <Form.Item
+                      name="client_secret"
+                      label={t('user_profile.sso.form.client_secret')}
+                      rules={[{ required: false }]}
+                      tooltip={t('user_profile.sso.form.client_secret_tooltip')}
+                    >
+                      <Input
+                        disabled={!isWriteable}
+                        style={{ width: 320 }}
+                        placeholder="********"
+                      />
+                    </Form.Item>
+                  )
+                }
+                {
+                  // we can uncomment this if we really need it
+                  /* <Form.Item
+                    name="scopes"
+                    label={t('user_profile.sso.form.scopes')}
+                    rules={[{ required: false }]}
+                  >
+                    <Input
+                      disabled={!isWriteable}
+                      style={{width: 320}}
+                      placeholder="openid profile email"
+                    />
+                  </Form.Item> */
+                }
```

Comment:
hi @horus , in your case, do you think you really need to customize `scopes` field?

Reply:
Yes, otherwise I wouldn't bother.

Talking about scopes, `profile` and `email` are all [optional](https://openid.net/specs/openid-connect-core-1_0.html#ScopeClaims), mind you. The original source got them hardcoded, since they're well-supported I don't think there'll be serious problems.

In some(that is, my) situation, one could use the scopes to implement access control, as I've already stated in the previous PR. These are fairly common and straight-forward, I let them there for the flexibility and/or future-proof.

Hope this clarifies.

Reply:
Ok, got it, I will update it later.